File: examples/integrations/node/rabbitmq_publish_agent_update.js
// Publish an agent update message to RabbitMQ
// Usage: node rabbitmq_publish_agent_update.js

const amqp = require('amqplib');

const QUEUE = process.env.AGENT_UPDATES_QUEUE || 'agent_updates';
const RABBIT_URL = process.env.RABBITMQ_URL || 'amqp://guest:guest@localhost:5672/';

(async () => {
  try {
    const conn = await amqp.connect(RABBIT_URL);
    const ch = await conn.createChannel();
    await ch.assertQueue(QUEUE, { durable: true });

    const cid = process.env.TEST_CONVERSATION_ID || 'demo-conv';
    const payload = {
      conversation_id: cid,
      event: 'agent:update',
      status: 'running',
      agent_id: 'agent-demo',
      data: { progress: 0.2, ts: Math.floor(Date.now() / 1000) },
    };

    const body = Buffer.from(JSON.stringify(payload));
    ch.sendToQueue(QUEUE, body, { contentType: 'application/json', persistent: true });
    console.log(`Published to ${QUEUE}:`, payload);
    await ch.close();
    await conn.close();
  } catch (err) {
    console.error('Publish failed:', err);
    process.exit(1);
  }
})();

File: examples/integrations/python/rabbitmq_publish_agent_update.py
import os
import json
import time

try:
    import pika  # type: ignore
except Exception as e:
    raise SystemExit("pika is required: pip install pika")

QUEUE = os.getenv("AGENT_UPDATES_QUEUE", "agent_updates")
RABBIT_URL = os.getenv("RABBITMQ_URL", "amqp://guest:guest@localhost:5672/")


def main():
    params = pika.URLParameters(RABBIT_URL)
    connection = pika.BlockingConnection(params)
    channel = connection.channel()
    channel.queue_declare(queue=QUEUE, durable=True)

    cid = os.getenv("TEST_CONVERSATION_ID", "demo-conv")
    payload = {
        "conversation_id": cid,
        "event": "agent:update",
        "status": "running",
        "agent_id": "agent-demo",
        "data": {"progress": 0.1, "ts": int(time.time())},
    }

    body = json.dumps(payload).encode("utf-8")
    channel.basic_publish(
        exchange="",
        routing_key=QUEUE,
        body=body,
        properties=pika.BasicProperties(content_type="application/json", delivery_mode=2),
    )
    print(f"Published to {QUEUE}: {payload}")
    connection.close()


if __name__ == "__main__":
    main()
